dali/dfu/dfurun.cpp

```diff
@@ -533,6 +533,24 @@ class CDFUengine: public CInterface, implements IDFUengine
         }
     }
 
+    void ensureFilePermissions(const char * fileName, IFileDescriptor * fd, SecAccessFlags perm, bool write)
+    {
+        if ((write && !HASWRITEPERMISSION(perm)) || (!write && !HASREADPERMISSION(perm)))
+        {
+            StringBuffer traceName;
+            if (isEmptyString(fileName))
```

Comment:
I'm not keen on the function passing 2 parameters one or both being able to serve up what's required.
I think better to simplify and,pass fileName only, and if seems worth it, put the code to get the tracename and truncate in a separate utility function - used in contexts that require it:
```
StringBuffer &getFDescName(const IFileDescriptor *fd, StringBuffer &result)
{
    fd->getTraceName(result);
    elideString(result, 255);
}
```
```
...
..
ensureFilePermissions(getFDescName(name), perm, write);
```

given u can express this in a couple of lines, not sure it's worth the extra helper function though..

Reply:
ok

Reply:
Should we check 'if ((write && !HASWRITEPERMISSION(perm)) || (!write && !HASREADPERMISSION(perm)))' before ensureFilePermissions(getFDescName(name), perm, write); ?

Reply:
If yes, the name ensureFilePermissions should be changed to throwFilePermissionException? If no, the getFDescName() will be called even if not denied.

Reply:
>If yes, the name ensureFilePermissions should be changed to throwFilePermissionException? If no, the getFDescName() will be called even if not denied.

I can live with it being called unconditionally, readability is more important than 100% efficiency here.
i.e. leave it as ensureFilePermissions and call getFDescName before hand (or the 2 lines of equivalent code)

Leave this PR for now though, let's get https://github.com/hpcc-systems/HPCC-Platform/pull/17563 agreed on and merged 1st.

Reply:
Now, I am not sure that it is worth to change the IFileDescriptor * fd to const char * fileName in the ensureFilePermissions(). Maybe leave the checkForeignFilePermissions() as it is (not call the ensureFilePermissions())?

Reply:
I suspect it is worth changing and calling from checkForeignFilePermissions, so still exercising common code.

Reply:
ok